contracts/oracles/MakerDAOOracleV2.sol

```
@@ -0,0 +1,126 @@
+        medianizer = IMedianizer(_medianizer);
+        currencyAddress = _currencyAddress;
+        currencySymbol = _currencySymbol;
+        polymathRegistry = IPolymathRegistry(_polymathRegistry);
+    }
+
+    /**
+      * @notice Updates medianizer address
+      * @param _medianizer Address of Maker medianizer
+      */
+    function changeMedianier(address _medianizer) public onlyOwner {
+        require(_medianizer != address(0), "0x not allowed");
+        /*solium-disable-next-line security/no-block-members*/
+        emit ChangeMedianizer(_medianizer, address(medianizer));
+        medianizer = IMedianizer(_medianizer);
+    }
+
+    /**
+    * @notice Returns address of oracle currency (0x0 for ETH)
+    */
+    function getCurrencyAddress() external view returns(address) {
+        return currencyAddress;
+    }
+
+    /**
+    * @notice Returns symbol of oracle currency (0x0 for ETH)
+    */
+    function getCurrencySymbol() external view returns(bytes32) {
+        return currencySymbol;
+    }
+
+    /**
+    * @notice Returns denomination of price
+    */
+    function getCurrencyDenominated() external view returns(bytes32) {
+        // All MakerDAO oracles are denominated in USD
+        return bytes32("USD");
+    }
+
+    /**
+    * @notice Returns price - should throw if not valid
+    */
+    function getPrice() external returns(uint256) {
+        _isCallerAllowed(msg.sender);
+        if (manualOverride) {
+            return manualPrice;
+        }
+        (bytes32 price, bool valid) = medianizer.peek();
+        require(valid, "MakerDAO Oracle returning invalid value");
+        return uint256(price);
+    }
+
+    // This function allows following contracts to read data from the oracles
+    // - STR Address
+    // - Any valid module of polymath ecosystem
+    function _isCallerAllowed(address _caller) view internal {
```

very minor, but I'd rename to _checkCallerAllowed if we are reverting on failure inside the function (rather than returning a bool)